src/App.js

```diff
@@ -35,21 +37,76 @@ const styles = theme => ({
 });
 
 class App extends Component {
+  constructor(props) {
```

Comment:
Let's put all of this logic in its own component, for example a `AuthenticationProvider` component that just renders its children

Reply:
Maybe `AuthProvider` is nicer. Shorter at least...